units/container_secret_cleanup.go

```diff
@@ -0,0 +1,119 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/evergreen-ci/cocoa"
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+const (
+	containerSecretCleanupJobName = "container-secret-cleanup"
+)
+
+func init() {
+	registry.AddJobType(containerSecretCleanupJobName, func() amboy.Job {
+		return makeContainerSecretCleanupJob()
+	})
+}
+
+type containerSecretCleanupJob struct {
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+
+	env       evergreen.Environment
+	smClient  cocoa.SecretsManagerClient
+	vault     cocoa.Vault
+	tagClient cocoa.TagClient
+}
+
+func makeContainerSecretCleanupJob() *containerSecretCleanupJob {
+	j := &containerSecretCleanupJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    containerSecretCleanupJobName,
+				Version: 0,
+			},
+		},
+	}
+	return j
+}
+
+// NewContainerSecretCleanupJob creates a job that cleans up stranded container
+// secrets that are not tracked by Evergreen.
+func NewContainerSecretCleanupJob(id string) amboy.Job {
+	j := makeContainerSecretCleanupJob()
+	j.SetID(fmt.Sprintf("%s.%s", containerSecretCleanupJobName, id))
+	return j
+}
+
+func (j *containerSecretCleanupJob) Run(ctx context.Context) {
+	defer func() {
+		j.MarkComplete()
+
+		if j.smClient != nil {
+			j.AddError(errors.Wrap(j.smClient.Close(ctx), "closing Secrets Manager client"))
+		}
+		if j.tagClient != nil {
+			j.AddError(errors.Wrap(j.tagClient.Close(ctx), "closing tag client"))
+		}
+	}()
+	if err := j.populate(); err != nil {
+		j.AddError(err)
+		return
+	}
+
+	secretIDs, err := cloud.GetFilteredResourceIDs(ctx, j.tagClient, []string{cloud.SecretsManagerResourceFilter}, map[string][]string{
+		cloud.PodCacheTag: {strconv.FormatBool(false)},
+	}, j.env.Settings().PodLifecycle.MaxSecretCleanupRate)
+	if err != nil {
+		j.AddError(errors.Wrap(err, "getting stranded Secrets Manager secrets"))
+		return
+	}
+
+	catcher := grip.NewBasicCatcher()
+	for _, secretID := range secretIDs {
+		catcher.Wrapf(j.vault.DeleteSecret(ctx, secretID), "secret '%s'", secretID)
+	}
+
+	j.AddError(errors.Wrap(catcher.Resolve(), "deleting secrets"))
+}
+
+func (j *containerSecretCleanupJob) populate() error {
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+
+	if j.smClient == nil {
```

Comment:
Is there a reason the secrets manager and tag clients are getting initialized independently of the vault (unlike [here](https://github.com/evergreen-ci/evergreen/blob/main/units/pod_allocator.go#L209)) in the pod allocator job?

Reply:
Hmm yeah, would be good to initialize them the same way. I'm going to change the pod allocator one to follow this because it's equivalent and slightly less if logic to think through.